dbms/src/Functions/FunctionsConversion.h

```diff
@@ -796,42 +924,30 @@ class FunctionConvert : public IFunction
             UInt64 scale = extractToDecimalScale(arguments[1]);
 
             if constexpr (std::is_same_v<Name, NameToDecimal32>)
-                return createDecimal(9, scale);
+                return createDecimal<DataTypeDecimal>(9, scale);
             else if constexpr (std::is_same_v<Name, NameToDecimal64>)
-                return createDecimal(18, scale);
+                return createDecimal<DataTypeDecimal>(18, scale);
             else if constexpr (std::is_same_v<Name, NameToDecimal128>)
-                return createDecimal(38, scale);
+                return createDecimal<DataTypeDecimal>(38, scale);
 
             throw Exception("Someting wrong with toDecimalNN()", ErrorCodes::LOGICAL_ERROR);
         }
         else
         {
-            /** Optional second argument with time zone is supported:
```

Comment:
We have lost this comment after refactoring. But it is useful.

Reply:
Revived the comment